tests/data/error_collision_geometry_not_exist.urdf

```diff
@@ -0,0 +1,37 @@
+<robot name="error_check">
```

Comment:
Can we reduce each of these test files to the minimal dateset to repro the expected issue?

For this one I believe none of the materials are relevant, nore is BaseLink, nor the `visual` in `link2`

Please apply this philosophy to all the test data. Only include the minimum dataset so it is clear the purpose of the file. Then make the robot name reflect that purpose e.g. `missing_geometry` rather than `error_check`